### PR TITLE
Clarify $SHELL guidance

### LINKED ISSUE #31: Don't suggest using $SHELL
This currently suggests in [the Environment Variables section](https://github.com/cli-guidelines/cli-guidelines/blob/main/content/_index.md#environment-variables-environment-variables):

> SHELL, if you’re going to invoke external commands.

Speaking with my fish-shell developer hat on, this causes us quite a few problems.

Specifically, some vim plugins use vim's `system()` function, which defers to $SHELL, to run some shellscript they supply, which breaks when the user has `chsh`'d to fish and it's not fish-compatible.

$SHELL reflects the user's preferred login shell and does absolutely not state what scripting language it uses. It could be bash or zsh, but also csh or elvish or fish or xonsh (or AFAIK /bin/false if the user is not supposed to be logged in as and you've `su`'d) or even python.

As POSIX states:

>    This variable shall represent a pathname of the user's preferred command language interpreter. If this interpreter does not conform to the Shell Command Language in the Shell and Utilities volume of IEEE Std 1003.1-2001, Chapter 2, Shell Command Language, utilities may behave differently from those described in IEEE Std 1003.1-2001.

The guidance should be changed to something like the following

- SHELL, which reflects the user's preferred login shell. Only use this when you want to drop the user into their shell or run some script they supplied. When you want to run some of your script, run it with /bin/sh or whatever other shell you have written it for.

### PR DESCRIPTION
**Pull request by faho**
Fixes #31

### CODE REVIEW

**Comment by aanand** [comment]
Thanks! This change makes sense to me. Two things:

1. We do one sentence per line to make diffs nicer.
2. Could we make the wording a bit more explicit?

Something like:

```markdown
- `SHELL`, if you need to open up an interactive terminal using the user's preferred shell program.
  (If you need to execute a shell script, use a specific interpreter like `/bin/sh`.)
```

File: content/_index.md [hunk 1]
```diff
@@ -1066,7 +1066,7 @@ Here’s a [list of POSIX standard env vars](https://pubs.opengroup.org/onlinepu
 - `EDITOR`, if you need to prompt the user to edit a file or input more than a single line.
 - `HTTP_PROXY`, `HTTPS_PROXY`, `ALL_PROXY` and `NO_PROXY`, if you’re going to perform network operations.
   (The HTTP library you’re using might already check for these.)
-- `SHELL`, if you’re going to invoke external commands.
+- `SHELL`, if you need to give the user their preferred shell. For scripts you wrote use /bin/sh or a more specific shell.
```
Comment (tashian) [inline content/_index.md:1069]:
```suggestion
- `SHELL`, if you need to give the user their preferred shell. For scripts you wrote, use `/bin/sh` or a more specific shell.
```

**Comment by tashian** [comment]
Thanks @faho! Just one suggestion here & otherwise it looks great.

**Comment by faho** [comment]
"shell program" sounds super weird to me.

How about

```
- `SHELL`, if you need to open up an interactive session of the user's preferred shell.
  (If you need to execute a shell script, use a specific interpreter like `/bin/sh`)
```

**Pull request merged**